ehcache-impl/src/main/java/org/ehcache/impl/persistence/CustomObjectInputStream.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.impl.persistence;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InvalidClassException;
+import java.io.ObjectStreamClass;
+import java.util.function.Predicate;
+
+public class CustomObjectInputStream extends CustomLoaderFileBasedObjectInputStream {
```

Comment:
Yes, we need this to decide the order of class loader used for class loading.
**Scenario 1 :**
    class loader passed as null --> 
        1. TCCL - It will look for TCCL
        2. Cache Manager - If not loaded from TCCL it will look into Cache Manager to load the classes
        3. latestUserDefinedLoader - If not loaded from Cache Manager it will use the Latest user defined(sun.misc.VM.latestUserDefinedLoader())

**Scenario 2 :** 
    Passing Custom class loader--> 
        1. Custom class loader - It will look for Custom class loader
        2. Cache Manager - If not loaded from Custom class loader it will look into Cache Manager to load the classes
        3. latestUserDefinedLoader - If not loaded from Cache Manager it will use the Latest user defined(sun.misc.VM.latestUserDefinedLoader())

Reply:
No I don't see any logic of classloading in this class, it is just doing the filtering as whether the class being loaded is permitted or not.
The class resolution is only being done in its super class, so the question remains - DO we need this kind if filtering for Filebased state repository.

Reply:
This kind predicate filtering is not required as class loading order logic is in parent class CustomLoaderFileBasedObjectInputStream. Hence We will remove CustomObjectInputStream class.